### ISSUE TITLE
Support EF Core 6

### ISSUE THREAD

**Issue by bricelam** [efcore/EFCore.FSharp#112]
Howdy! We're starting to wind down on the EF Core 6 release, and I wanted to reach out to see if you needed any support from us.

The big new language-dependent design-time changes I can remember are:

* Scaffolding many-to-many relationships (already tracked by #109)
* Temporal tables (just need to support a [new pattern](https://github.com/dotnet/efcore/pull/24562/files#diff-d464e59a794ae7fd54b1770c49a469cbf1fe00850256d8660bd0fc833cc70a6e) of NestedClosureCodeFragment)
* Compiled models (you're actually probably blocked on https://github.com/dotnet/efcore/issues/25110)

**Comment by simon-reynolds** [comment]
Hi @bricelam 
I've started work on EF Core 6.0 support, have NestedClosureCodeFragment support, and working on the many-to-many relationships

I also want to use the changes to ValueConverters to support null (dotnet/efcore#13850) to allow translating nullable database columns to Option types

I'll see what I can do for compiled models, creating an implementation `FSharpRuntimeModelCodeGenerator` might take a while. If dotnet/efcore#25110 could be a blocker then I'll see what I can hack my way around.....
